spec/features/comments_spec.rb

```diff
@@ -39,11 +44,29 @@
     end
   end
 
+  describe "inaccessible comment form" do
+    let!(:article) { create(:article) }
+
+    it "should not let a user access comment form" do
+      visit article_path(article)
+
+      expect(page).to have_no_content("Logout")
+      expect(page).to have_no_content("Add your comment here")
+      
+      expect(page).to have_content "Please login to comment."
+      expect(page).to have_content "Login"
+    end
+  end
+
   describe "comment counter" do
+    before do
+      page.set_rack_session(userinfo: { "info" => { "name" => "Paul"}})
```

Comment:
You could, if you wanted to be fancy, put this into a helper called `login_user` or something and then just call that method in the before block rather than using this line of code each time.